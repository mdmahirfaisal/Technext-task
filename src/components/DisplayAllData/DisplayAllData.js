import * as React from 'react';
import { styled } from '@mui/material/styles';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Collapse from '@mui/material/Collapse';
import Avatar from '@mui/material/Avatar';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import MoreVertIcon from '@mui/icons-material/MoreVert';
import YouTubeIcon from '@mui/icons-material/YouTube';
import ArticleIcon from '@mui/icons-material/Article';
import RocketLaunchIcon from '@mui/icons-material/RocketLaunch';


const ExpandMore = styled((props) => {
    const { expand, ...other } = props;
    return <IconButton {...other} />;
})(({ theme, expand }) => ({
    transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
    marginLeft: 'auto',
    transition: theme.transitions.create('transform', {
        duration: theme.transitions.duration.shortest,
    }),
}));

const DisplayAllData = ({ launch }) => {
    const { details, links, rocket, flight_number, launch_date_local, launch_date_unix, launch_failure_details, launch_site, launch_success, launch_year, video_link, mission_name, static_fire_date_unix, upcoming, tentative_max_precision, reddit_campaign, } = launch;

    const externalLinkOpen = (link) => {
        console.log(link)
    }

    // expand control
    const [expanded, setExpanded] = React.useState(false);
    const handleExpandClick = () => {
        setExpanded(!expanded);
    };

    return (
        <Card>
            <CardHeader sx={{ textAlign: 'start' }}
                avatar={
                    <Avatar aria-label="recipe">
                        <img className='mx-auto' src={links?.mission_patch_small} alt="Avatar" />
                    </Avatar>
                }
                action={
                    <IconButton aria-label="settings">
                        <MoreVertIcon />
                    </IconButton>
                }
                title={mission_name}
                subheader={launch_date_local}
            />
            <img className='h-[150px] md:h-[200px] mx-auto' src={links?.mission_patch} alt="Mission img" />
            <CardContent sx={{ textAlign: 'start' }}>
                <Typography variant="body2">
                    <span className='font-bold'>Rocket: </span> {rocket?.rocket_name || "Rocket name is missing"}
                </Typography>
                <Typography variant="body2" >
                    <span className='font-bold'>Site name: </span> {launch_site?.site_name || "No Details Available"}
                </Typography>
            </CardContent>

            {/* ---- External Links ---- */}

            <CardActions disableSpacing>

                <a href={links?.article_link} target="_blank" rel="noopener noreferrer">
                    <IconButton  >
                        <ArticleIcon />
                    </IconButton></a>
                <a href={links?.reddit_launch} target="_blank" rel="noopener noreferrer">
                    <IconButton  >
                        <RocketLaunchIcon />
                    </IconButton></a>
                <a href={links?.video_link} target="_blank" rel="noopener noreferrer">
                    <IconButton  >
                        <YouTubeIcon />
                    </IconButton></a>

                <ExpandMore
                    expand={expanded}
                    onClick={handleExpandClick}
                    aria-expanded={expanded}
                    aria-label="show more"
                >
                    <ExpandMoreIcon />
                </ExpandMore>
            </CardActions>

            <Collapse in={expanded} timeout="auto" unmountOnExit>
                <CardContent sx={{ textAlign: 'start' }}>

                    <Typography paragraph color="text.secondary">
                        <span className='font-bold'>Details: </span> {details || "No Details Available"}
                    </Typography>

                </CardContent>
            </Collapse>
        </Card>
    );
}

export default DisplayAllData;