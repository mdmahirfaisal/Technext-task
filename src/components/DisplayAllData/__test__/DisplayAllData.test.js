import React from 'react'
import ReactDOM from 'react-dom'
import DisplayAllData from '../DisplayAllData'
import renderer from 'react-test-renderer'
import "@testing-library/jest-dom";
import { Provider } from 'react-redux';
import { store } from '../../../redux/store';


it('display all data render without crashing',
    () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <Provider store={store}>
                <DisplayAllData />
            </Provider>
            , div)
    })


it("matches display all data snapshot",
    () => {
        const tree = renderer.create(
            <Provider store={store}>
                <DisplayAllData />
            </Provider>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })