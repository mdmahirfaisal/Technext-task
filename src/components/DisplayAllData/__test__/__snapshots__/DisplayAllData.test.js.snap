// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches display all data snapshot 1`] = `
<div
  className="rounded-tl-[70px] rounded-br-[70px] bg-white py-10"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardHeader-root css-19cxssf-MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        <div
          aria-label="recipe"
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-2s90m6-MuiAvatar-root"
        >
          <img
            alt="Avatar"
            className="mx-auto"
          />
        </div>
      </div>
      <div
        className="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          className="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
        >
          Launch Date:  undefined
        </span>
      </div>
    </div>
    <img
      alt="Mission img"
      className="h-[200px] md:h-[250px] mx-auto"
    />
    <div
      className="MuiCardContent-root css-besep6-MuiCardContent-root"
    >
      <h2
        className="text-xl md:text-2xl font-medium md:font-semibold "
      >
        Rocket name is not available
      </h2>
      <p
        className="absolute right-5 top-5 text-lg"
      >
        Failure
      </p>
      <div
        className="mt-2"
      >
        <p>
           
          <span
            className="font-medium text-red-600"
          >
            Flight number: 
          </span>
           
        </p>
        <p>
           
          <span
            className="font-medium text-red-600"
          >
            Launch year: 
          </span>
           
        </p>
        <p>
            
          <span
            className="font-medium text-red-600"
          >
            Nationality: 
          </span>
        </p>
        <p>
          <span
            className="font-medium text-red-600"
          >
            Site name: 
          </span>
           
          No Details Available
        </p>
        <p>
           
          <span
            className="font-medium text-red-600"
          >
            Landing vehicle: 
          </span>
        </p>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-15yr02a-MuiSvgIcon-root"
          data-testid="ArticleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
          />
        </svg>
      </button>
    </a>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1jogx4x-MuiSvgIcon-root"
          data-testid="LinkIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3.9 12c0-1.71 1.39-3.1 3.1-3.1h4V7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h4v-1.9H7c-1.71 0-3.1-1.39-3.1-3.1zM8 13h8v-2H8v2zm9-6h-4v1.9h4c1.71 0 3.1 1.39 3.1 3.1s-1.39 3.1-3.1 3.1h-4V17h4c2.76 0 5-2.24 5-5s-2.24-5-5-5z"
          />
        </svg>
      </button>
    </a>
    <a
      rel="noopener noreferrer"
      target="_blank"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1idedao-MuiSvgIcon-root"
          data-testid="YouTubeIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 15l5.19-3L10 9v6m11.56-7.83c.13.47.22 1.1.28 1.9.07.8.1 1.49.1 2.09L22 12c0 2.19-.16 3.8-.44 4.83-.25.9-.83 1.48-1.73 1.73-.47.13-1.33.22-2.65.28-1.3.07-2.49.1-3.59.1L12 19c-4.19 0-6.8-.16-7.83-.44-.9-.25-1.48-.83-1.73-1.73-.13-.47-.22-1.1-.28-1.9-.07-.8-.1-1.49-.1-2.09L2 12c0-2.19.16-3.8.44-4.83.25-.9.83-1.48 1.73-1.73.47-.13 1.33-.22 2.65-.28 1.3-.07 2.49-.1 3.59-.1L12 5c4.19 0 6.8.16 7.83.44.9.25 1.48.83 1.73 1.73z"
          />
        </svg>
      </button>
    </a>
    <button
      aria-expanded={false}
      aria-label="show more"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1hzudc3-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="ExpandMoreIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
        />
      </svg>
    </button>
  </div>
</div>
`;
